antlr_config/ManimParser.g4

```diff
@@ -4,35 +4,52 @@ options {
-    | IDENT EQUAL expr                                              #AssignmentStatement
-    | method_call                                                   #MethodCallStatement
-    | RETURN expr                                                   #ReturnStatement;
+stat: SLEEP OPEN_PARENTHESIS expr CLOSE_PARENTHESIS ';'                 #SleepStatement
+    | COMMENT OPEN_PARENTHESIS STRING CLOSE_PARENTHESIS ';'             #CommentStatement // when string type defined we can adjust
+    | LET IDENT (COLON type)? EQUAL expr ';'                            #DeclarationStatement
+    | IDENT EQUAL expr ';'                                              #AssignmentStatement
+    | IF '(' ifCond=expr ')' '{' ifStat=stat? '}'
+     elseIf*
+    (ELSE '{' elseStat=stat? '}')?                                      #IfStatement
+    | stat1=stat stat2=stat                                             #ConsecutiveStatement
+    | method_call ';'                                                   #MethodCallStatement
+    | RETURN expr ';'                                                   #ReturnStatement;
+
+
+elseIf: ELSE IF '(' elifCond=expr ')' '{' elifStat=stat? '}'*;
 
 arg_list: expr (COMMA expr)*                                        #ArgumentList;
 
 expr: NUMBER                                                        #NumberLiteral
+    | bool                                                          #BooleanLiteral
     | IDENT                                                         #Identifier
     | NEW data_structure_type                                       #DataStructureContructor
     | method_call                                                   #MethodCallExpression
-    | unary_operator=(ADD | MINUS) expr                             #UnaryOperator
-    | expr binary_operator=(ADD | MINUS | TIMES) expr               #BinaryExpression;
+    | unary_operator=(ADD | MINUS | NOT) expr                       #UnaryOperator
+    | left=expr binary_operator=(ADD | MINUS | TIMES) right=expr    #BinaryExpression
+    | left=expr binary_operator=(GT | GE | LE | LT) right=expr      #BinaryExpression
+    | left=expr binary_operator=(EQ | NEQ) right=expr               #BinaryExpression
+    | left=expr binary_operator=(AND | OR) right=expr               #BinaryExpression
+    ;
 
 method_call: IDENT DOT IDENT OPEN_PARENTHESIS arg_list? CLOSE_PARENTHESIS  #MethodCall
            | IDENT OPEN_PARENTHESIS arg_list? CLOSE_PARENTHESIS            #FunctionCall;
 
 type: data_structure_type                                            #DataStructureType
     | primitive_type                                                 #PrimitiveType;
 
-data_structure_type: STACK OPEN_GENERIC primitive_type CLOSE_GENERIC #StackType;
-primitive_type: NUMBER_TYPE                                          #NumberType;
+data_structure_type: STACK '<' primitive_type '>'                    #StackType;
```

Comment:
Did you want to remove these from the lexer grammar for any particular reason? I feel like it's better practice to have all of the tokens in the lexer file: https://stackoverflow.com/a/2498247

Reply:
They are in the lexer file

Reply:
Why is it not using `OPEN_GENERIC` and `CLOSE_GENERIC` any more?

Reply:
Awkward naming since semantically they can be used in this case and as an operator

Reply:
It's also more readable personally

Reply:
I renamed them to GT and LT

Reply:
I'm still not sure since our parser file is now quite inconsistent. We have literals like '(' and '{' and also lexer tokens like `OPEN_PARENTHESIS`. I'll wait to see what other people think. @mayanksharma3 @adityagoel4512 @claralebbos @iz0825

Reply:
I understand your point about '<' and '>' being used in different contexts (for data structures and for boolean expressions) but it should be consistent throughout

Reply:
Sure that sounds good.

Reply:
Personally I think it would be better if it's consistent? (We can do either direct '(' or OPEN_PARENTHESIS but it might be better if we keep it consistent)

Reply:
Yeah same I think it should be consistent too, they should all be as tokens in the lexer file, as a pose to directly doing '>'

Reply:
They have to be tokens in the lexer file to do it directly anyways